### PR TITLE
refactor: rewrite test result computing

### PR DESCRIPTION
**Pull request by ondrej-hosak**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -12,5 +12,11 @@ Metrics/LineLength:
 Metrics/MethodLength:
   Max: 17
 
+Metrics/ClassLength:
```
Comment (panjan) [inline .rubocop.yml:15]: rubocop fails anyway so no need to increase class length

File: lib/test_aggregation/build_result.rb [hunk 2]
```diff
@@ -80,40 +41,82 @@ def find_job(job_id)
...
+      # when 'loading' step exists
+      return 'Created' if job_results.include?('created')
+
+      'Passed'
+    end
+
+    def machine_result(results)
+      return 'Passed' if results.all? { |r| ['Passed', 'Skipped', 'NotPerformed'].include?(r) }
+
+      return 'Failed' if results.include? 'Failed' # test on 810
```
Comment (panjan) [inline lib/test_aggregation/build_result.rb]: interesting comment

File: lib/test_aggregation/build_result.rb [hunk 3]
```diff
@@ -123,6 +126,24 @@ def results_hash(opts = {})
       res
     end
 
+    def new_states_results_hash
```
Comment (panjan) [inline lib/test_aggregation/build_result.rb:129]: a comment would be nice here

Comment (petrSchreiber) [inline lib/test_aggregation/build_result.rb:129]: comment

File: lib/test_aggregation/build_result.rb [hunk 4]
```diff
@@ -155,7 +177,7 @@ def machines_as_json(part_name, machine_name)
                end # What to do when zero or more jobs is defined?
       {
         os: machine_name,
-        result: result(part: part_name, machine: machine_name),
+        result: jobs.first.state == 'canceled' ? 'Failed' : jobs.first.state.capitalize,
```
Comment (panjan) [inline lib/test_aggregation/build_result.rb:180]: should the result really be jobs.first.state?

File: lib/test_aggregation/step_result.rb [hunk 5]
```diff
@@ -74,7 +75,31 @@ def results_hash(opts = {})
...
+      all_machines = {
+        'all' => {
+          result: build_result.machine_result(all_states),
+          message: '',
+          resultId: '00000000-0000-0000-0000-000000000000'
+        }
+      }
+
+      {
+        id: __id__,
```
Comment (panjan) [inline lib/test_aggregation/step_result.rb]: why object id?

File: spec/test_aggregation/build_result_spec.rb [hunk 6]
```diff
@@ -1,96 +1,117 @@
...
-        'job_id'          => 1,
-        'name'            => 'step name',
-        'position'        => 1,
-        'class_name'      => 'class name',
-        'class_position'  => 1
-      }
+    it 'returns Failed when there are canceled jobs' do
+      step_data.merge!(
+        'job_id' => 985,
+        'result' => 'created'
```
Comment (panjan) [inline spec/test_aggregation/build_result_spec.rb:86]: in-consistence created vs. canceled

File: spec/test_aggregation/class_result_spec.rb [hunk 7]
```diff
@@ -27,10 +28,10 @@
     end
 
     it 'raise exception when class_name differs' do
```
Comment (panjan) [inline spec/test_aggregation/class_result_spec.rb]: raises

File: spec/test_aggregation/build_result_spec.rb (hunk 6 above)
Comment (ondrej-hosak) [inline spec/test_aggregation/build_result_spec.rb:86]: its step status not job status. Step status canceled does not exists.

File: lib/test_aggregation/build_result.rb (hunk 4 above)
Comment (ondrej-hosak) [inline lib/test_aggregation/build_result.rb:180]: yes, ask @h4rr0d for more information

**Pull request merged**